src/coldcard.js

```diff
@@ -0,0 +1,115 @@
+/**
+ * Provides classes for interacting with a Coldcard via TXT/JSON/PSBT files
+ *
+ * The following API classes are implemented:
+ *
+ * * ColdcardExportExtendedPublicKey
+ * * ColdcardSignMultisigTransaction
+ * * ColdcardMultisigWalletConfig
+ *
+ * @module coldcard
+ */
+import {
+  IndirectKeystoreInteraction,
+} from "./interaction";
+
+export const COLDCARD = 'coldcard';
+export const WALLET_CONFIG_VERSION = "0.0.1";
+
+
+/**
+ * Base class for indirect interactions with Coldcard.
+ *
+ * @extends {module:interaction.IndirectKeystoreInteraction}
+ */
+export class ColdcardInteraction extends IndirectKeystoreInteraction {
+
+}
+
+/**
+ * Base class for coldcard interactions (reading/writing files)
+ *
+ * @extends {module:hermit.HermitInteraction}
+ */
+export class ColdcardFileReader extends ColdcardInteraction {
+
+}
+
+export class ColdcardExportExtendedPublicKey extends ColdcardFileReader {
+
+}
+
+/**
+ * Returns signature request data to display via PSBT for Coldcard and returns the PSBT
+ * data back to the calling application
+ */
+export class ColdcardSignTransaction extends ColdcardFileReader {
+
+}
+
+/**
+ * Returns a valid multisig wallet config text file to send over to a Coldcard
+ *
+ * NOTE: only the root xfp of the signing device is required to be correct, but we
+ * recommend only setting up the multisig wallet on the Coldcard with complete xfp
+ * information. We define isSupported() to check exactly this condition.
+ *
+ * This is an example Coldcard config file from
+ * https://coldcardwallet.com/docs/multisig
+ *
+ * # Coldcard Multisig setup file (exported from 4369050F)
+ * #
+ * Name: MeMyself
+ * Policy: 2 of 4
+ * Derivation: m/45'
+ * Format: P2WSH
+ *
+ * D0CFA66B: tpubD9429UXFGCTKJ9NdiNK4rC5...DdP9
+ * 8E697B74: tpubD97nVL37v5tWyMf9ofh5rzn...XgSc
+ * BE26B07B: tpubD9ArfXowvGHnuECKdGXVKDM...FxPa
+ * 4369050F: tpubD8NXmKsmWp3a3DXhbihAYbY...9C8n
+ *
+ */
+export class ColdcardMultisigWalletConfig {
+  constructor({jsonConfig}) {
+    if (jsonConfig &&
+      (jsonConfig.uuid || jsonConfig.name) &&
```

Comment:
This is partly just a personal taste thing, but I think it's more readable to do these conditionals one at a type (unless they are dependent on each other).

i.e. 
```
if (jsonConfig.name && jsonConfig.uuid)
  this.name = jsonConfig.uuid || jsonConfig.name
```
An added benefit of this is that if there are any specific properties that are absolutely required vs some that aren't, you can throw in that conditional and say what's missing.

Reply:
Also for this section, do you have any feelings on also including type checking? Even though JS obviously isn't strictly typed, you can add asserts that will throw to make sure everything is kosher. This also makes this initialization a very effective validator pretty much out of the gate (which you want anyway since the class is useless if any field is not formatted correctly).